### PR TITLE
chore(ci): add github action for labelling pull requests based on changed files

### PR DESCRIPTION
**Pull request by atzoum**
## Description of the change

The following labels will be automatically added:

1. `with tests` whenever a test file is added/updated
2. `server-team` whenever go source is added/updated in paths other than `/warehouse`
3. `warehouse-team` whenever files are added/updated inside the `/warehouse` path

## Notion Link

https://www.notion.so/rudderstacks/Add-github-action-for-labelling-pull-requests-based-on-changed-files-24baa33202d943cfa7c11542e2bfeac4

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+test:
```
Comment (lvrach) [inline .github/labeler.yml:1]: I was thinking of something more self-explanatory
`with tests` or `contains tests`

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+- any: ['**/*_test.go']
+
+server:
```
Comment (lvrach) [inline .github/labeler.yml:4]:
```suggestion
server-team:
```

File: .github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+- any: ['**/*.go']
+  all: ['!warehouse/**/*']
+
+warehouse:
```
Comment (lvrach) [inline .github/labeler.yml:8]:
```suggestion
warehouse-team:
```

File: .github/labeler.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+- any: ['warehouse/**/*']
```
Comment (chrikar) [inline .github/labeler.yml:9]: There's many more we could add here but this is a really good starting point to check out how this looks without putting too many labels.

**Review by lvrach — COMMENTED** [review]
Some suggestions, lets get feedback from more people on this

**Review by mcchran — APPROVED** [review]
LGTM.

**Comment by mcchran** [comment]
I like labeld PRs.

**Review by lvrach — APPROVED** [review]

**Pull request merged**